storage/src/adb/any/variable/mod.rs

```diff
@@ -661,12 +676,21 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Codec, H: CHasher, T: Translato
             .await?;
         self.uncommitted_ops = 0;
 
+        // Persist the oldest retained location that we will prune to before actually pruning
+        // the MMR and operations log. If we crash after writing this but before pruning, we will
+        // recover on restart by recognizing that the actual oldest retained location we find in
+        // the log is in a different section than the one we persisted here.
+        let target_prune_loc = self.target_prune_loc();
```

Comment:
I think we may want to call the same thing in `close`?

Reply:
I actually don't think we need to update the metadata in `close`. The only time we need to update and write `metadata` is in `commit` before we `self.sync()`. We could even remove the `self.metadata.sync()` from `self.sync()` -- it should always be a no-op since we write the metadata before calling `sync()` in `commit()`